data/mods/gen6/random-teams.ts

```diff
@@ -1,46 +1,33 @@
-'use strict';
+import {TeamData} from '../../random-teams';
+import RandomGen7Teams from '../gen7/random-teams';
+import {PRNG, PRNGSeed} from '../../../sim/prng';
+import {FS} from '../../../lib/fs';
 
-const RandomGen7Teams = require('../gen7/random-teams');
-
-class RandomGen6Teams extends RandomGen7Teams {
-	/**
-	 * @param {Format | string} format
-	 * @param {?PRNG | [number, number, number, number]} [prng]
-	 */
-	constructor(format, prng) {
+export class RandomGen6Teams extends RandomGen7Teams {
+	constructor(format: Format | string, prng: PRNG | PRNGSeed | null) {
 		super(format, prng);
-		/**@type {AnyObject} */
-		// @ts-ignore
-		this.randomFactorySets = require('./factory-sets.json');
+		this.randomFactorySets = JSON.parse(FS('./factory-sets.json').readIfExistsSync() || "{}");
```

Comment:
why though????????

Reply:
First: `FS` is relative to PS root.

Second: This code silently fails if the file doesn't exist (causing line 981 to crash), instead of giving an error message like `require` would.

This is a really important point: **Error messages are good because they tell you what's wrong.** Suppressing errors is bad unless the error condition is something that's actually intended to be supported.